### PR TITLE
Create doc before ending server response

### PR DESCRIPTION
**Pull request by jvwong**
This was noticed on the demo (Editor) page, but updates the server so that POST request to `api/document` now end server response  after the doc is created. 

Refs #676.

### CODE REVIEW

File: src/server/routes/api/document/index.js [hunk 1]
```diff
@@ -544,10 +544,10 @@ http.post('/', function( req, res, next ){
   };
 
   checkRequestContext( provided )
-    .then( () => res.end() )
     .then( () => createSecret({ secret }) )
     .then( loadTables )
     .then( handleDocCreation )
+    .then( doc => { res.end(); return doc; } )
```
Comment (maxkfranz) [inline src/server/routes/api/document/index.js]: Should something like this be at the very end (just before `next`)?  Maybe we should use [`sendStatus()`](http://expressjs.com/en/5x/api.html#res.sendStatus) instead

Comment (maxkfranz) [reply src/server/routes/api/document/index.js]: Or maybe the response should be the document JSON

Comment (jvwong) [reply src/server/routes/api/document/index.js]: I'll return JSON as this is part of my upcoming PR for #675 where the doc link is provided in the success for CTA.

Comment (jvwong) [reply src/server/routes/api/document/index.js]: Updated to return JSON at end, but you might notice it's a bit laggy (2-3s to end response)

Comment (maxkfranz) [reply src/server/routes/api/document/index.js]: Is it the email part creating most of the lag?

Comment (jvwong) [reply src/server/routes/api/document/index.js]: Yes, that's my best guess just having observed the server console logging

Comment (maxkfranz) [reply src/server/routes/api/document/index.js]: Maybe that part could be after the sendStatus().  The main thing is that the document JSON is stable before sending off the http response.

Comment (jvwong) [reply src/server/routes/api/document/index.js]: Yes I think that's the solution.

Comment (maxkfranz) [reply src/server/routes/api/document/index.js]: Sounds good

**Pull request merged**